src/exoscale/compute/api/http.clj

```diff
@@ -108,8 +108,10 @@
                                                     :pagesize pagesize))
                (fn [resp]
                  (let [acc (concat acc resp)
-                       all-results-present? (= (:count (meta resp)) (count resp))]
-                   (if (or single-page-only?
+                       meta-count (:count (meta resp))
+                       all-results-present? (= meta-count (count acc))]
```

Comment:
Fix - we now compare the count of the accumulated results, not of the response, this also caused us to recur indefinitely.